orderer/common/channelparticipation/restapi.go

```diff
@@ -9,21 +9,36 @@ package channelparticipation
 import (
 	"encoding/json"
 	"net/http"
+	"path"
+	"strings"
 
+	"github.com/gorilla/mux"
+	"github.com/hyperledger/fabric/common/configtx"
 	"github.com/hyperledger/fabric/common/flogging"
 	"github.com/hyperledger/fabric/orderer/common/localconfig"
+	"github.com/hyperledger/fabric/orderer/common/types"
 	"github.com/pkg/errors"
 )
 
-const URLBaseV1 = "/participation/v1/"
-
-type ErrorResponse struct {
-	Error string `json:"error"`
-}
+const (
+	URLBaseV1           = "/participation/v1/"
+	URLBaseV1Channels   = URLBaseV1 + "channels"
+	channelIDKey        = "channelID"
+	urlWithChannelIDKey = URLBaseV1Channels + "/{" + channelIDKey + "}"
+)
 
 //go:generate counterfeiter -o mocks/channel_management.go -fake-name ChannelManagement . ChannelManagement
 
 type ChannelManagement interface {
+	// ListAllChannels returns a slice of ChannelInfoShort containing all application channels (excluding the system
+	// channel), and ChannelInfoShort of the system channel (nil if does not exist).
+	// The URL fields are empty, and are to be completed by the caller.
+	ListAllChannels() types.ChannelList
```

Comment:
I thought we agreed to rename APIs here?

Reply:
Actually we didn't, we just agreed on the return types, see here:
https://github.com/hyperledger/fabric/pull/1232#discussion_r423875968

Reply:
I recall you have confirmed at least twice:

https://github.com/hyperledger/fabric/pull/1232#discussion_r422969830
https://github.com/hyperledger/fabric/pull/1232#discussion_r422969632

Reply:
I change ChannelInfoLong to ChannelInfo, and used the tuple suggestion as well, but not the name of the method...

Reply:
done